wrappers/desktop.go

```diff
@@ -138,16 +129,10 @@ func sanitizeDesktopFile(s *snap.Info, desktopFile string, rawcontent []byte) []
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		// whitespace/comments are just copied
-		if strings.TrimSpace(line) == "" || strings.HasPrefix(strings.TrimSpace(line), "#") {
-			newContent = append(newContent, line)
+		if !isValidDesktopFileLine(line) {
```

Comment:
Just a nitpick, could we please log those lines (e.g: `ignoring line %d in desktop file %s`). This could be done in a separate branch if logging from this place is inappropriate.